src/http.rs

```diff
@@ -1,15 +1,40 @@
 use crate::repository::MessageRepository;
 use hyper::http::HeaderValue;
-use hyper::service::{make_service_fn, service_fn};
-use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use hyper::{Body, Request, Response, Server, StatusCode};
 use log::info;
-use regex::Regex;
+use routerify::ext::RequestExt;
+use routerify::{Router, RouterService};
+use routerify_cors::enable_cors_all;
 use serde::Serialize;
-use std::convert::Infallible;
+use std::io;
 use std::net::TcpListener;
-use std::str::FromStr;
 use std::sync::{Arc, Mutex};
 
+struct State {
+    repository: Arc<Mutex<MessageRepository>>,
+}
+
+fn router(repository: Arc<Mutex<MessageRepository>>) -> Router<Body, io::Error> {
+    let state = State { repository };
+
+    Router::builder()
+        .middleware(enable_cors_all())
```

Comment:
To be clear, I suggest replacing `enable_cors_all()` with `Middleware::post(add_cors_headers)` and adding this `add_cors_headers` function:

```
async fn add_cors_headers(mut res: Response<Body>) -> Result<Response<Body>, io::Error>
{
    let headers = res.headers_mut();

    headers.insert(header::ACCESS_CONTROL_ALLOW_ORIGIN, HeaderValue::from_static("*"));
    headers.insert(header::ACCESS_CONTROL_ALLOW_METHODS, HeaderValue::from_static("*"));
    headers.insert(header::ACCESS_CONTROL_ALLOW_HEADERS, HeaderValue::from_static("*"));
    headers.insert(header::ACCESS_CONTROL_EXPOSE_HEADERS, HeaderValue::from_static("*"));

    Ok(res)
}
```

I always try to avoid extremely small dependencies and I feel routerify-cors is not worth adding as a dependency. Can you please make these changes?

Reply:
@pevdh I updated the PR! Thanks